app/tests/algorithms_tests/test_models.py

```diff
@@ -89,3 +89,21 @@ def test_outputs_are_set():
     assert len(outputs) == 1
     assert outputs[0].interface.kind == InterfaceKindChoices.JSON
     assert outputs[0].value == {"dsaf": 35421}
+
+    job = AlgorithmJobFactory()
+    job.algorithm_image.algorithm.result_template = (
+        "foo score: {{result_dict.foo}}"
+    )
+
+    assert job.rendered_result_text == ""
+    job.create_result(result={"foo": 13.37})
+    assert job.rendered_result_text == "<p>foo score: 13.37</p>"
+
+    job.algorithm_image.algorithm.result_template = "{% for key, value in dict.metrics.items() -%}{{ key }}  {{ value }}{% endfor %}"
+    assert (
+        job.rendered_result_text
+        == "Jinja template is incorrect: 'type object' has no attribute 'metrics'"
+    )
```

Comment:
```suggestion
    assert job.rendered_result_text == "Jinja template is invalid"
```
If you accept the changes abvoe